LINQPractice/Services/EmployeeServices.cs

```diff
@@ -0,0 +1,85 @@
+﻿using LINQPractice.Contracts;
+using LINQPractice.Data;
+using LINQPractice.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+namespace LINQPractice.Services
+{
+    public class EmployeeServices : IEmployeeServices
+    {
+        PrjContext context;
+        public EmployeeServices(PrjContext Prjcontext)
+        {
+
+            context = Prjcontext;
+        }
+        public async Task<IEnumerable<Employee>> Get()
+        {
+            var employees = await context.Employees.ToListAsync();
+            return employees;
+        }
+
+        public async Task<IEnumerable<Department>> GetD()
+        {
+            var employees = await context.Departments.ToListAsync();
+            return employees;
+        }
+        public async Task<IEnumerable<Employee>> EmployeesByDepartment(int Id)
+        {
+            var employees = await context.Employees.Where(e => e.DepartmentId == Id).ToListAsync();
+            return employees;
+        }
+
+        public async Task<int> EmployeesCount()
+        {
+            var employeeCount = context.Employees.Count();
+            return employeeCount;
+        }
+        public async Task<double> GetAvgNameLength()
```

Comment:
This async method lacks await, I think you are warned by sonar here. Please check if possible.